### ISSUE TITLE
Unable to stop activity {com.ventrata.app/im.crisp.client.ChatActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.content.Context.getSystemService(java.lang.String)' on a null object reference

### ISSUE THREAD

**Issue by FrantisekGazo** [crisp-im/crisp-sdk-android#73]
I found multiple exceptions like this coming from the crisp sdk.

crisp sdk version: 1.0.7

```
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.content.Context.getSystemService(java.lang.String)' on a null object reference
    at im.crisp.client.b.d.d.b.i
    at im.crisp.client.b.d.d.b.b
    at im.crisp.client.b.e.b.a.onStop
    at androidx.fragment.app.Fragment.performStop(Fragment.java:2894)
    at androidx.fragment.app.FragmentStateManager.stop(FragmentStateManager.java:391)
    at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1211)
    at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356)
    at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434)
    at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1497)
    at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2625)
    at androidx.fragment.app.FragmentManager.dispatchStop(FragmentManager.java:2598)
    at androidx.fragment.app.FragmentController.dispatchStop(FragmentController.java:291)
    at androidx.fragment.app.FragmentActivity.onStop(FragmentActivity.java:563)
    at androidx.appcompat.app.AppCompatActivity.onStop(AppCompatActivity.java:216)
    at android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1277)
    at android.app.Activity.performStop(Activity.java:6185)
    at android.app.ActivityThread.handleSleeping(ActivityThread.java:3501)
    at android.app.ActivityThread.access$2800(ActivityThread.java:152)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1479)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:135)
    at android.app.ActivityThread.main(ActivityThread.java:5259)
    at java.lang.reflect.Method.invoke(Method.java)
    at java.lang.reflect.Method.invoke(Method.java:372)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:902)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:697)

java.lang.RuntimeException: Unable to stop activity {com.ventrata.app/im.crisp.client.ChatActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.content.Context.getSystemService(java.lang.String)' on a null object reference
    at android.app.ActivityThread.handleSleeping(ActivityThread.java:3504)
    at android.app.ActivityThread.access$2800(ActivityThread.java:152)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1479)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:135)
    at android.app.ActivityThread.main(ActivityThread.java:5259)
    at java.lang.reflect.Method.invoke(Method.java)
    at java.lang.reflect.Method.invoke(Method.java:372)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:902)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:697)
```

and like this (same but in `ActivityThread.handleDestroyActivity`)
```
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.content.Context.getSystemService(java.lang.String)' on a null object reference
    at im.crisp.client.b.d.d.b.i
    at im.crisp.client.b.d.d.b.b
    at im.crisp.client.b.e.b.a.onStop
    at androidx.fragment.app.Fragment.performStop(Fragment.java:2894)
    at androidx.fragment.app.FragmentStateManager.stop(FragmentStateManager.java:391)
    at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1211)
    at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1356)
    at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:1434)
    at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1497)
    at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:2625)
    at androidx.fragment.app.FragmentManager.dispatchStop(FragmentManager.java:2598)
    at androidx.fragment.app.FragmentController.dispatchStop(FragmentController.java:291)
    at androidx.fragment.app.FragmentActivity.onStop(FragmentActivity.java:563)
    at androidx.appcompat.app.AppCompatActivity.onStop(AppCompatActivity.java:216)
    at android.app.Instrumentation.callActivityOnStop(Instrumentation.java:1277)
    at android.app.Activity.performStop(Activity.java:6185)
    at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:3670)
    at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:3729)
    at android.app.ActivityThread.access$1400(ActivityThread.java:152)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1358)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:135)
    at android.app.ActivityThread.main(ActivityThread.java:5259)
    at java.lang.reflect.Method.invoke(Method.java)
    at java.lang.reflect.Method.invoke(Method.java:372)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:902)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:697)

java.lang.RuntimeException: Unable to stop activity {com.ventrata.app/im.crisp.client.ChatActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.content.Context.getSystemService(java.lang.String)' on a null object reference
    at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:3675)
    at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:3729)
    at android.app.ActivityThread.access$1400(ActivityThread.java:152)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1358)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:135)
    at android.app.ActivityThread.main(ActivityThread.java:5259)
    at java.lang.reflect.Method.invoke(Method.java)
    at java.lang.reflect.Method.invoke(Method.java:372)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:902)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:697)
```

**Comment by eliottvincent** [comment]
Thanks, we'll take a look at those exceptions and what's causing them.

**Comment by Doc1faux** [comment]
Hi @FrantisekGazo and thank your for your feedback :)
The issue has been addressed in 1.0.9.
I let you reopen the issue if you face it again.
